src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -365,6 +367,7 @@ private async Task StopAsyncCore(bool disposing)
 
                 CheckDisposed();
                 connectionState = _connectionState;
+                await taskQueue.Drain();
```

Comment:
Shouldn't we change states before we start draining? It also doesn't make sense to store away the local `connectionState` before a potentially long await.

Also, we should at the very least log when draining takes a very long time.